### PR TITLE
Update links to Jenkins and add links to github repos in READMEs

### PR DESCRIPTION
**Pull request by leenagupte**
Add links in the documentation to make it easier to find things referenced in documentation (e.g. Jenkins job and an application repositories).

### CODE REVIEW

**Comment by floehopper** [comment]
Minor: Ideally this would be split into 3 commits - one for each line in the PR description. As it stands the commit note subject is not a complete summary of the changes in that commit.

**Comment by floehopper** [comment]
Minor: It's worth noting that this documentation may be out-of-date. There has been some related discussion in this [GitHub issue][1] and this [Trello card][2]. It might be worth mentioning this either in the PR description or the relevant commit note - otherwise the changes might give the impression that the documentation *is* up-to-date!

[1]: https://github.com/alphagov/smart-answers/issues/2356
[2]: https://trello.com/c/leHXvtqN

File: doc/deploying.md [hunk 1]
```diff
@@ -2,13 +2,13 @@
...
 
-This happens automatically as a post build action when the govuk_smart_answers build passes on CI.
+This happens automatically as a post build action when the [govuk_smart_answers](https://ci.dev.publishing.service.gov.uk/job/govuk_smart_answers/) build passes on CI.
 
 ### Manually
 
 If you have sufficient permissions, you can use the [Integration Deploy App](https://deploy.integration.publishing.service.gov.uk/job/Deploy_App/). Click the "Build with Parameters" link and enter "release" as the "TAG".
 
-If you do not have sufficient permissions, you can either kick the main CI build in Jenkins or commit & push something to the `master` branch.
+If you do not have sufficient permissions, you can either kick the [main CI build](https://github.com/alphagov/smart-answers/blob/master/doc/continuous-integration.md#main) in Jenkins or commit & push something to the `master` branch.
```
Comment (floehopper) [inline doc/deploying.md]: Minor: We may not have been consistent about this elsewhere, but I think links to other docs would be better as relative links i.e. `[main CI build](https://github.com/alphagov/smart-answers/blob/master/doc/continuous-integration.md#main)` -> `[main CI build](continuous-integration.md#main)`.

**Comment by floehopper** [comment]
Other than my minor comments, this looks good to me.

File: doc/deploying.md (hunk 1 above)
Comment (leenagupte) [inline doc/deploying.md]: @floehopper I agree, and have made the change. I check the rest of the code base and there aren't any other absolute links to the READMEs in the docs folder.

**Comment by leenagupte** [comment]
@floehopper Thanks for comments. I'll keep the commits separate next time.

**Comment by floehopper** [comment]
@leenagupte: Cool. @chrisroos and I often [use git to curate the commits][1] in a PR before merging them to make them easier to follow. I know this PR is relatively trivial, but I unless you're really pressed for time, I would still have a go at (a) combining the 2nd commit into the 1st one (as if you'd used a relative link first time round); and (b) split up the resulting commit into 3 commits as I suggested in my previous comment. Let me know if you're unsure how to do this and I can try to help.

[1]: https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

**Comment by leenagupte** [comment]
@floehopper I've split the commit into three. One per file updated.

**Comment by floehopper** [comment]
@leenagupte: Looks great. Thanks.

**Comment by floehopper** [comment]
I just thought I'd mention that @chrisroos & I usually try to rebase a branch against `master` before merging it.

I think we started doing this so that the git history is easier to follow (i.e. more linear), but I think another advantage is that if you push the rebased branch to the GitHub remote (either force-push or push a new version of the branch), then the `govuk_smartanswers_branches` CI build will run against your changes in combination with the latest changes in `master`. This should avoid scenarios where although a branch merges cleanly from git's (textual) point of view, the build fails due to a [semantic conflict][1].

I realise this is something else that isn't documented anywhere, so I've added a [Trello card][2] to write this up.

I know this PR is just documentation, so in this case a semantic conflict should not occur.

[1]: http://martinfowler.com/bliki/SemanticConflict.html
[2]: https://trello.com/c/oMV0lG8v

**Comment by leenagupte** [comment]
@floehopper I don't have access to the Trello board.

**Pull request merged**